zebedee-cms/src/main/java/com/github/onsdigital/zebedee/service/ZebedeeDatasetService.java

```diff
@@ -51,27 +53,39 @@ public CollectionDataset updateDatasetInCollection(String collectionID, String d
         }
 
         Dataset dataset = datasetClient.getDataset(datasetID);
-        collectionDataset.setTitle(dataset.getTitle());
-        collectionDataset.setUri(dataset.getLinks().getSelf().getHref());
-
-        if (dataset.getState().equals(State.CREATED)) {
-            // the dataset has just been added to the collection, so update collection ID on the dataset
-            Dataset datasetUpdate = new Dataset();
-            datasetUpdate.setCollection_id(collectionID);
-            datasetUpdate.setState(State.ASSOCIATED);
-            datasetClient.updateDataset(datasetID, datasetUpdate);
-        }
 
-        if (dataset.getState().equals(State.ASSOCIATED)
-                && !dataset.getCollection_id().equals(collectionID)) {
-            throw new ConflictException("cannot add dataset " + datasetID
-                    + " to collection " + collectionID
-                    + " it is already in collection " + dataset.getCollection_id());
+        if (dataset != null) {
+
+            collectionDataset.setTitle(dataset.getTitle());
+
+            if (dataset.getLinks() != null && dataset.getLinks().getSelf() != null) {
+                collectionDataset.setUri(dataset.getLinks().getSelf().getHref());
+            } else {
+                logInfo("The dataset URL has not been set on the dataset response.")
+                        .addParameter("collectionID", collectionID)
+                        .addParameter("datasetID", datasetID)
+                        .log();
```

Comment:
Should this throw an exception? So The HTTP handler can send the correct code?

Reply:
Yes you are right. I had been ignoring this case while a bug gets fixed on the dataset API, but its correct to just throw the exception if its not there. 👍